lib/auth/bot_test.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth/native"
+	"github.com/gravitational/teleport/lib/modules"
+	"github.com/gravitational/teleport/lib/tlsca"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/ssh"
+)
+
+type botTestModules struct {
+	modules.Modules
+}
+
+func (m *botTestModules) Features() modules.Features {
+	return modules.Features{
+		MachineID: false, // Explicitly turn off Machine ID
+	}
+}
+
+func TestBotCreateFeatureDisabled(t *testing.T) {
+	srv := newTestTLSServer(t)
+
+	defaultModules := modules.GetModules()
+	defer modules.SetModules(defaultModules)
+	modules.SetModules(&botTestModules{})
```

Comment:
We have a `modules.SetTestModules(t, m)` that will automatically restore the modules on cleanup. I would use it here.

Reply:
Nice! Didn't know that existed, but it's much nicer.